### ISSUE TITLE
Better handling of circular reference within tasks

### ISSUE THREAD

**Issue by fliespl** [deployphp/deployer#2496]
I am aware that it might be edge case, but such error might be hard to find. It makes sense to handle such mis-uses with exception (circular reference exception).

Currently it ends up with "Segmentation fault" which could mean plenty of things.

- Deployer version: 7.0beta19
- Deployment OS: Debian/Ubuntu

```
task('deploy:test2', [
   'deploy:test'
]);

task('deploy:test', [
    'deploy:test2'
]);

desc('Deploy project');
task('deploy', [
    'deploy:test',
    'deploy:publish',
]);
```

**Comment by antonmedv** [comment]
Nice catch. Will fix it.

**Comment by antonmedv** [comment]
Detection implemented in d6c484bc4162d47c39a1c5b09f43b06b633c7f4d.

**Comment by fliespl** [comment]
Great! Wroks as expected :)
